src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,22 +49,101 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block + ", " + street + ", " + unit + ", " + postalCode;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.equals(((Address) other).block)
+                && this.street.equals(((Address) other).street)
+                && this.unit.equals(((Address) other).unit)
+                && this.postalCode.equals(((Address) other).postalCode)); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
+    
+    class Block {
+    	private int block;
+
+		public Block(int block) {
+			super();
+			this.block = block;
+		}
+		public int getBlock() {
+			return block;
+		}
+		public void setBlock(int block) {
+			this.block = block;
+		}
+		@Override
+		public String toString() {
+			return Integer.toString(block);
+		}
+    }
+    
+    class Street {
+    	private String street;
+
+		public Street(String street) {
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.